spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/EnvironmentEntryInitializingTreeMap.java

```diff
@@ -90,4 +90,8 @@ public T put(String key, T value) {
 		return delegate.entrySet();
 	}
 
+	@Override
+	public boolean containsKey(Object key) {
```

Comment:
As a note: as the javadoc clearly states, we do not intend to use this as a general purpose data structure, but as a workaround until https://github.com/spring-projects/spring-boot/issues/7986 is fixed (only in Spring Boot 2, unfortunately). 

That being said, your solution is valid too in the context. Thanks for the contribution!